### PR TITLE
feat: Add Golang-based transformer for Ruby

### PR DESCRIPTION
**Pull request by Akash-Nayak**
Signed-off-by: Akash Nayak <akash19nayak@gmail.com>

### CODE REVIEW

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+// RubyDockerfileGenerator implements the Transformer interface
+type RubyDockerfileGenerator struct {
+	Config       transformertypes.Transformer
+	Env          *environment.Environment
+	NodejsConfig NodejsDockerfileYamlConfig
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:35]: Shouldn't it be Ruby?

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+// RubyTemplateConfig implements Nodejs config interface
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:38]: Shouldn't the comment be Ruby

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *RubyDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *RubyDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	Gemfiles, err := common.GetFilesByName(dir, []string{"", "Gemfile"}, nil)
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:63]: Why is there a empty string

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	if err != nil {
+		logrus.Debugf("Cannot get the Gemfile: %s", err)
+		return nil, nil, nil
+	}
+	if len(Gemfiles) > 0 {
+		rubyFiles, err := common.GetFilesByExt(dir, []string{".rb"})
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:69]: Do you want to make .rb a constant

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+			logrus.Errorf("Error while finding ruby files %s", err)
+		}
+		var serviceName string
+		if len(rubyFiles) == 1 {
+			serviceName = strings.TrimSuffix(filepath.Base(rubyFiles[0]), ".rb")
+		}
+		if serviceName != "" {
+			namedServices = map[string]transformertypes.ServicePlan{
+				serviceName: []transformertypes.TransformerPlan{{
+					Mode:              t.Config.Spec.Mode,
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:80]: Mughe be better to create the transformer plan once and use it in both the places depending on whether it is named or unnamed

File: transformer/classes/dockerfile/rubydockerfilegenerator.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+			Type:     transformertypes.SourcePathMappingType,
+			DestPath: common.DefaultSourceDir,
+		}, transformertypes.PathMapping{
+			Type:           transformertypes.TemplatePathMappingType,
+			SrcPath:        filepath.Join(t.Env.Context, t.Config.Spec.TemplatesDir),
+			DestPath:       filepath.Join(common.DefaultSourceDir, relSrcPath),
+			TemplateConfig: rubyConfig,
+		})
+		paths := a.Paths
+		paths[artifacts.DockerfilePathType] = []string{filepath.Join(common.DefaultSourceDir, relSrcPath, "Dockerfile")}
```
Comment (ashokponkumar) [inline transformer/classes/dockerfile/rubydockerfilegenerator.go:144]: I think we have a constant for Dockerfile string.

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Thanks. Few comments

File: transformer/classes/dockerfile/rubydockerfilegenerator.go (hunk 6 above)
Comment (Akash-Nayak) [reply transformer/classes/dockerfile/rubydockerfilegenerator.go]: Yes, I have changed the hard-coded string to the Dockerfile string.

File: transformer/classes/dockerfile/rubydockerfilegenerator.go (hunk 5 above)
Comment (Akash-Nayak) [reply transformer/classes/dockerfile/rubydockerfilegenerator.go]: Yes sure, done.

File: transformer/classes/dockerfile/rubydockerfilegenerator.go (hunk 2 above)
Comment (Akash-Nayak) [reply transformer/classes/dockerfile/rubydockerfilegenerator.go]: Yes, I have changed it now.

File: transformer/classes/dockerfile/rubydockerfilegenerator.go (hunk 1 above)
Comment (Akash-Nayak) [reply transformer/classes/dockerfile/rubydockerfilegenerator.go]: Yes, it should be Ruby. I missed it during the find and replace. It’s my bad.

File: transformer/classes/dockerfile/rubydockerfilegenerator.go (hunk 3 above)
Comment (Akash-Nayak) [reply transformer/classes/dockerfile/rubydockerfilegenerator.go]: Removed the empty string.

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**